src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -21,6 +21,8 @@ object OpenHashMap {
   def apply[K, V](elems : (K, V)*) = new OpenHashMap[K, V] ++= elems
   def empty[K, V] = new OpenHashMap[K, V]
 
+  private val MaximumTableCapacity = 1 << 30
```

Comment:
Why don't you ever check that the _new_ size isn't greater than this?  If you start with the default of 8 = 1<<3, and the table increases by *4 each time, you'll jump straight from 1<<29 to 1<<31.